src/Moq/Mock.cs

```diff
@@ -548,6 +548,41 @@ internal static MethodCall SetupSet(Mock mock, LambdaExpression expression, Cond
 			return Mock.Setup(mock, expression, condition);
 		}
 
+		// This specialized version of `SetupSet` exists to let `Mock.Of` support properties that are not overridable.
+		// Note that we generally prefer having a setup for a property's return value, but in this case, that isn't possible.
+		internal static void SetupSet(Mock mock, LambdaExpression expression, PropertyInfo propertyToSet, object value)
+		{
+			Guard.NotNull(expression, nameof(expression));
+
+			Mock.SetupRecursive<MethodCall>(mock, expression, setupLast: (targetMock, _, __) =>
+			{
+				// For strict mocks, we haven't called `SetupAllProperties` on the mock being set up.
+				// Therefore, whenever a property is being initialized, we quickly need to enable auto-stubbing.
+				//
+				// (One would think that it would be simpler to perform `SetupAllProperties` at the beginning
+				// and leave it enabled until the initialized mock is returned to the user. However, transforming
+				// the LINQ query such that a final disable of auto-stubbing happens is much more difficult!)
+
+				var temporaryAutoSetupProperties = targetMock.AutoSetupPropertiesDefaultValueProvider == null;
+				if (temporaryAutoSetupProperties)
+				{
+					targetMock.AutoSetupPropertiesDefaultValueProvider = targetMock.DefaultValueProvider;
+				}
+				try
+				{
+					propertyToSet.SetValue(targetMock.Object, value, null);
```

Comment:
Perhaps we could go one step further here and add support for indexers by passing through the index arguments instead of hard-coding `null` as the last argument.

Reply:
Leaving this unresolved for now. Supporting indexers for non-overridable properties would only work partly anyway, e.g. you could only use fixed values (no matchers) for the index arguments. Not sure if it's worth doing the work to support this scenario as far as is possible.